frame/assets/src/lib.rs

```diff
@@ -902,8 +910,14 @@ pub mod pallet {
 		) -> DispatchResult {
 			let origin = ensure_signed(origin)?;
 
-			ensure!(name.len() <= T::StringLimit::get() as usize, Error::<T, I>::BadMetadata);
-			ensure!(symbol.len() <= T::StringLimit::get() as usize, Error::<T, I>::BadMetadata);
+			let bounded_name: BoundedVec<u8, T::StringLimit> = name
```

Comment:
due to the usage of bounded vec here we should make sure that the limit is not exceeded before migration.
Also any modification of `T::StringLimit` should ensure the on chain items don't exceed the limit.

If these consideration are too strong, we should move to WeakBoundedVec.

Also we could use BoundedVec for the call argument directly instead of Vec + conversion here. But I think this can be done later, as UI needs to handle the new type.